### PR TITLE
Control datastore via REST API

### PR DESCRIPTION
**Pull request by dafeder**
This adds an endpoint and resources for controling a _resource_ node's datastore via the REST API.

## QA Steps

Using whatever means you prefer to test the API (I like Postman; [here is a collection that includes requests for all these QA steps](https://gist.github.com/dafeder/be673ea8e51845bf3a09645e2a718e97)), test the following actions, using the example of the "Violent Crime Statistics By City" resource:

- [ ] Create datastore configuration: POST /api/dataset/datastore/19:

```json
{
	"manager": "simple",
	"configuration": {
		"delimiter": ",",
		"quote": "\"",
		"escape": "\\"
	} 
}
```

- [ ] Retrieve datastore information: GET /api/dataset/datastore/19

- [ ] Update datastore configuration: PUT /api/dataset/datastore/19: 

```json
{
	"configuration": {
		"delimiter": "|",
		"quote": "\"",
		"escape": "\\"
	} 
}
```
(Also try enabling the fast import module and changing the `manager` property to `fast`; it should refuse)

- [ ] Remove datastore: DELETE /api/dataset/datastore/19
- [ ] Import datastore: POST /api/dataset/datastore/19/import
- [ ] Drop datastore: POST /api/dataset/datastore/19/drop

### CODE REVIEW

**Comment by dafeder** [comment]
@fmizzell I am a little confused about why the resource actions cause an import immediately rather than adding to the queue. I don't think that's an issue with the services code, though.

File: modules/dkan/dkan_datastore/resources/dkan_datastore_resource.inc [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+    services_error("The datastore for Resource {$resource_nid} has not been configured.");
+  }
+  return $datastore->getStatus();
+}
+
+function _dkan_datastore_resource_create($resource_nid, $data) {
+  $data = _services_arg_value($data, 'data');
+  $manager = $data['manager'];
+  $configuration = $data['configuration'];
+  // return $configuration;
```
Comment (fmizzell) [inline modules/dkan/dkan_datastore/resources/dkan_datastore_resource.inc]: @dafeder I tiny remnant of the past :)

File: modules/dkan/dkan_datastore/resources/dkan_datastore_resource.inc [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+    }
+    return $datastore->getStatus();
+  }
+}
+
+function _dkan_datastore_resource_update($resource_nid, $data) {
+  $data = _services_arg_value($data, 'data');
+  $manager = $data['manager'];
+  $configuration = $data['configuration'];
+  // return $configuration;
```
Comment (fmizzell) [inline modules/dkan/dkan_datastore/resources/dkan_datastore_resource.inc]: Another one.

File: modules/dkan/dkan_datastore/resources/dkan_datastore_resource.inc [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+  }
+  return NULL;
+}
+
+function _dkan_datastore_resource_info() {
+  $info = dkan_datastore_managers_info();
+  return $info;
+}
+
+function _dkan_datastore_resource_access() {
```
Comment (fmizzell) [inline modules/dkan/dkan_datastore/resources/dkan_datastore_resource.inc]: Do we want to change this to use a level of node access?

File: modules/dkan/dkan_dataset/modules/dkan_dataset_rest_api/dkan_dataset_rest_api.info [hunk 4]
```diff
@@ -1,5 +1,5 @@
 name = DKAN Dataset REST API
-description = "REST APIs for DKAN Datasets"
+description = REST APIs for DKAN Datasets
 core = 7.x
```
Comment (fmizzell) [inline modules/dkan/dkan_dataset/modules/dkan_dataset_rest_api/dkan_dataset_rest_api.info]: We should make the dependency between this module and dkan_datastore explicit.

File: modules/dkan/dkan_datastore/resources/dkan_datastore_resource.inc (hunk 3 above)
Comment (dafeder) [reply modules/dkan/dkan_datastore/resources/dkan_datastore_resource.inc]: Doing in separate ticket

**Comment by kimwdavidson** [comment]
awaiting tests

**Comment by kimwdavidson** [comment]
Needs rebasing after #2481 and #2482

File: drupal-org.make [hunk 5]
```diff
@@ -23,6 +23,12 @@ projects:
     version: '2.3'
     patch:
       2833824: https://www.drupal.org/files/issues/autocomplete-deluxe-2833824-4.patch
+  autoload:
```
Comment (fmizzell) [inline drupal-org.make]: This is no longer needed

File: modules/dkan/dkan_datastore/modules/dkan_datastore_fast_import/psr-4/FastImport.php [hunk 6]
```diff
@@ -0,0 +1,66 @@
+<?php
```
Comment (fmizzell) [inline modules/dkan/dkan_datastore/modules/dkan_datastore_fast_import/psr-4/FastImport.php]: Is this a rebasing/merging issue?

File: modules/dkan/dkan_datastore/modules/dkan_datastore_simple_import/psr-4/SimpleImport.php [hunk 7]
```diff
@@ -0,0 +1,105 @@
+<?php
```
Comment (fmizzell) [inline modules/dkan/dkan_datastore/modules/dkan_datastore_simple_import/psr-4/SimpleImport.php]: Vestiges of autoload.

**Pull request merged**